docs/processed-profile-format.md

```diff
@@ -0,0 +1,8 @@
+# Processed profile format
+
+The Gecko Profiler emits JSON profiles that perf.html can display. This original Gecko profile format needs to be processed in order to provide a data structure that is optimized for JS analysis. This client-side step provides an easy mechanism for pre-processesing certain data transformations. The largest transformation is moving the performance data to be from a list of small object for each entry, to a few array tables containing long lists of single values. For instance if each profiler sample had a single object associated per sample with 7 properties per object, then 1000 samples would be one array containing 1000 objects. Transforming that into the processed format that same 1000 samples would be 1 table object, and 7 arrays, each of length 1000. The latter is much easier for the Garbage Collector to handle when using a reactive programming style.
```

Comment:
The important part is that the array entries are now primitive values (usually numbers or null) and no longer objects, and the GC doesn't need to be aware of primitive values.